### PR TITLE
Fixed some compiler warnings

### PR DESCRIPTION
**Pull request by kompjoefriek**
- signed/unsigned mismatch
- printing enum with %d
- using int as bool

### CODE REVIEW

**Comment by minggo** [comment]
Could you please replace tab with 4 spaces.

**Comment by kompjoefriek** [comment]
[ci rebuild]

@minggo should be OK now

**Comment by minggo** [comment]
Thanks. 
On which platform will have these warnings?

**Comment by kompjoefriek** [comment]
signed/unsigned mismatch --> Visual Studio 2013
using int as bool --> Visual Studio 2013
printing enum with %d --> Android NDK r10d x86_64 on Windows

**Comment by minggo** [comment]
Thanks.
Could you please fix conflict?

**Comment by kompjoefriek** [comment]
I tried to fix the conflict. It seems i'm not compatible with git.

The visual studio project files and Camera3DTest.cpp where not supposed to be changed, and my machine keeps changing the version of tools/cocos2d-console.
I don't know how to clean this pullrequest now...

Maybe i should get a fresh copy of v4 and make a new branch and submit a new pull request?

**Comment by minggo** [comment]
@kompjoefriek 
There is not conflict now.

File: tools/cocos2d-console [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit cd05722966edde6116a6096b810a7a5598fe7c0a
+Subproject commit 53130d33970b5d9b15a8f0e4c2632e352a6ade9f
```
Comment (minggo) [inline tools/cocos2d-console]: Please don't modify submodule.

**Comment by kompjoefriek** [comment]
@minggo 
It should be good to merge again.
I found that Github for windows will automatically update submodules on checkout...

**Comment by minggo** [comment]
@kompjoefriek 
You should update submodule when submodule is changed.

**Pull request merged**